README.markdown

```diff
@@ -370,7 +378,38 @@ An array of resources can also be passed in, and each will be created with the t
 
 *Type*: statement.
 
-#### `flatten`
+#### `ensure_resources`
+
+Takes a resource type, title (only hash), and a hash of attributes that describe the resource(s).
+
+~~~
+user { 'dan':
+  gid => 'mygroup',
+  ensure => present,
+}
+~~~
+
+An hash of resources should be passed in and each will be created with the type and parameters specified if it doesn't already exist:
+
+    `ensure_resources('user', {'dan' => { gid => 'mygroup', uid => '600' } ,  'alex' => { gid => 'mygroup' }}, {'ensure' => 'present'})`
+
+From Hiera Backend:
+
+  ~~~
+  userlist:
+    dan:
+      gid: 'mygroup'
+   uid: '600'
+    alex:
+   gid: 'mygroup'
+  ~~~
+
+Call:
+
+    `ensure_resources('user', $userlist, {'ensure' => 'present'})`
```

Comment:
I think this has to be `ensure_resources('user', hiera_hash('userlist'), {'ensure' => 'present'})`